include/parameter.h

```diff
@@ -0,0 +1,307 @@
+
+    static inline const std::string _desc = "";
+    static inline const std::string _name = "";
+
+    template<typename P> static constexpr T _default_t(decltype(P::_default) *) { return P::_default; }
+    template<typename P> static constexpr T _default_t(...) { return _default; }
+    template<typename P> static constexpr T _min_t(decltype(P::_min) *) { return P::_min; }
+    template<typename P> static constexpr T _min_t(...) { return _min; }
+    template<typename P> static constexpr T _max_t(decltype(P::_max) *) { return P::_max; }
+    template<typename P> static constexpr T _max_t(...) { return _max; }
+    template<typename P> static const std::string _desc_t(decltype(P::_desc) *) { return P::_desc; }
+    template<typename P> static const std::string _desc_t(...) { return _desc; }
+    template<typename P> static const std::string _name_t(decltype(P::_name) *) { return P::_name; }
+    template<typename P> static const std::string _name_t(...) { return _name; }
+
+    bool _free = true;
+    std::unique_ptr<Limiter> _limiter;
+    std::unique_ptr<Transformer> _transformer;
+
+    std::string _label;
+    std::unique_ptr<SetC> _inheritors = nullptr;
+    std::unique_ptr<SetC> _modifiers = nullptr;
+
+    std::shared_ptr<const Limits<T>> _limits_ptr;
+    std::shared_ptr<const Transform<T>> _transform_ptr;
+    std::shared_ptr<const Unit> _unit_ptr;
+    std::shared_ptr<const C> _inheritee_ptr = nullptr;
+
+    void _set_value(T value)
+    {
+        if (!(get_limits().check(value))) throw std::runtime_error(
+            "Value=" + std::to_string(value) + "beyond get_limits()=" + get_limits().str()
+        );
+        _value = value;
+    }
+
+protected:
+    T _value;
+    T _value_transformed;
+
+    void set_inheritee(std::shared_ptr<const C> inheritee) {
+        _inheritee_ptr = inheritee == nullptr ? nullptr : std::move(inheritee);
+    }
+
+public:
+    inline void add_inheritor(std::shared_ptr<C> inheritor) { 
+        if(inheritor == nullptr) throw std::runtime_error("Can't add null inheritor to " + this->str());
+        if(!inheritor->get_free()) "Can't add_inheritor(" + inheritor->str() + ") with fixed inheritor";
+        if(!inheritor->get_inheritors().empty()) throw std::runtime_error("Can't add_inheritor(" + 
+            inheritor->str() + ") with inheritors of its own; inheritance may not be nested");
+        if(_inheritors == nullptr) _inheritors = std::make_unique<SetC>();
+        _inheritors->insert(inheritor);
+    }
+    inline void add_modifier(std::shared_ptr<C> modifier) { 
+        if(modifier == nullptr) throw std::runtime_error("Can't add null modifier to " + this->str());
+        if(_modifiers == nullptr) _modifiers = std::make_unique<SetC>();
+        _modifiers->insert(modifier);
+    }
+
+    static const std::string get_desc() { return _desc_t<C>(0); }
+    static constexpr T get_default() { return _default_t<C>(0); }
+    bool get_fixed() const { return !_free; }
```

Comment:
```suggestion
    bool get_fixed() const override { return !_free; }
```
You should really do this on all of the inherited methods you override anywhere.  I bet there's a way to get your IDE to point them out.

Reply:
I haven't found such a way. Does VSCode warn you by default?

Reply:
I haven't noticed in VSCode, actually - I'm just extrapolating from something I'm pretty sure other people have told me that `clang-tidy` can do.